infrastructure/aws/index.sh

```diff
@@ -24,16 +24,36 @@ CONFIG_INPUT="$6"
 
 EC2_INSTANCE_ID=$(wget -q -O - http://instance-data/latest/meta-data/instance-id)
 
+
+echo "Creating index-scratch on local instance SSD"
+
+# Create the "index-scratch" directory where each specific tree's indexing
+# byproducts live while the indexing is ongoing.
+#
+# Under the old c2d image /dev/xvdb was the instance-locale storage and was
+# somehow already mounted as /mnt.  Under c5d it's /dev/nvme1n1 and wasn't
+# already mounted.  We just dynamically initialize it here.
+sudo mkfs -t ext4 /dev/nvme1n1
+sudo mount /dev/nvme1n1 /mnt
+sudo mkdir /mnt/index-scratch
+sudo chown ubuntu.ubuntu /mnt/index-scratch
+
+
 echo "Branch is $BRANCH"
 echo "Channel is $CHANNEL"
 
 export AWS_ROOT=$(realpath $MOZSEARCH_PATH/infrastructure/aws)
 VOLUME_ID=$(python $AWS_ROOT/attach-index-volume.py $CHANNEL $EC2_INSTANCE_ID)
 
+# The EBS volume will no longer be mounted at /dev/xvdf but instead at an
+# arbitrarily assigned nvme id.  However, since we only have a single EBS volume
+# and we dynamically attach it, we're pretty certain what the ID will be:
+EBS_NVME_DEV=nvme2n1
+
 set +o pipefail   # The grep command below can return nonzero, so temporarily allow pipefail
 while true
```

Comment:
We should change this `while true` to a bounded loop, `for ((i = 0; i < 3600; i++))` (or similar) so that if the device is called something else we won't spin forever.

Reply:
I've made this change and pushed to "dev" (plus the speculative change at https://bugzilla.mozilla.org/show_bug.cgi?id=1567724#c16).